### PR TITLE
Truncate block payloads in server logs

### PR DESCRIPTION
**Pull request by vxgmichel**
`block_create` requests and `block_read` replies have by far the largest payloads (up to 512K).

In debug mode, each call to those commands will add up to 1.5MB of content to the logs (due to ASCII representation).

Truncating those payloads allows to reduce the logs to a reasonable size.

### CODE REVIEW

File: server/parsec/asgi/rpc.py [hunk 1]
```diff
@@ -54,6 +54,32 @@
 logger = get_logger()
 
 
+def block_repr(block: bytes) -> str:
+    if len(block) <= 64:
+        return repr(block)
+    return f"{block[:64]}... ({len(block)} bytes)"
```
Comment (mmmarcos) [inline server/parsec/asgi/rpc.py:60]: Maybe make it more explicit?
```suggestion
MAX_LOGGED_BLOCK_SIZE = 64

def block_repr(block: bytes) -> str:
    """Truncate block repr to save space in logs"""
    if len(block) <= MAX_LOGGED_BLOCK_SIZE:
        return repr(block)
    return f"{block[:MAX_LOGGED_BLOCK_SIZE]}... ({len(block)} bytes)"
```

File: server/parsec/asgi/rpc.py [hunk 2]
```diff
@@ -54,6 +54,32 @@
...
+
+
+@dataclass
+class LoggedReq:
+    req: object
+
+    def __repr__(self) -> str:
+        if not isinstance(self.req, authenticated_cmds.latest.block_create.Req):
+            return repr(self.req)
+        return f"BlockCreateReq {{ block: {block_repr(self.req.block)} }}"
```
Comment (mmmarcos) [inline server/parsec/asgi/rpc.py:70]: Maybe redundant but how about adding the comments you put in the PR description? 
Also I would reverse condition (matter of taste ^^)
```suggestion
        # block_create has a large payload (up to 512K)
        # truncate it to save space in logs 
        if isinstance(self.req, authenticated_cmds.latest.block_create.Req):
            return f"BlockCreateReq {{ block: {block_repr(self.req.block)} }}"
        return repr(self.req)
```

File: server/parsec/asgi/rpc.py [hunk 3]
```diff
@@ -54,6 +54,32 @@
...
+
+
+@dataclass
+class LoggedRep:
+    rep: object
+
+    def __repr__(self) -> str:
+        if not isinstance(self.rep, authenticated_cmds.latest.block_read.RepOk):
+            return repr(self.rep)
+        return f"Ok {{ block: {block_repr(self.rep.block)}, key_index: {self.rep.key_index!r}, needed_realm_certificate_timestamp: {self.rep.needed_realm_certificate_timestamp!r} }}"
```
Comment (mmmarcos) [inline server/parsec/asgi/rpc.py:80]:
```suggestion
        # block_read has a large payload (up to 512K)
        # truncate it to save space in logs 
        if isinstance(self.rep, authenticated_cmds.latest.block_read.RepOk):
            return f"Ok {{ block: {block_repr(self.rep.block)}, key_index: {self.rep.key_index!r}, needed_realm_certificate_timestamp: {self.rep.needed_realm_certificate_timestamp!r} }}"
        return repr(self.rep)
```

**Review by mmmarcos — APPROVED** [review]
LGTM. Some suggestions to take or leave.

File: server/parsec/asgi/rpc.py (hunk 2 above)
Comment (vxgmichel) [reply server/parsec/asgi/rpc.py]: 4556e098267f7fb6060d38048074439c1100a41f

File: server/parsec/asgi/rpc.py (hunk 3 above)
Comment (vxgmichel) [reply server/parsec/asgi/rpc.py]: 4556e098267f7fb6060d38048074439c1100a41f

**Pull request merged**